engine/src/main/java/com/redhat/cloud/notifications/utils/CloudEventParser.java

```diff
@@ -0,0 +1,126 @@
+package com.redhat.cloud.notifications.utils;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.networknt.schema.ApplyDefaultsStrategy;
+import com.networknt.schema.JsonMetaSchema;
+import com.networknt.schema.JsonSchema;
+import com.networknt.schema.JsonSchemaException;
+import com.networknt.schema.JsonSchemaFactory;
+import com.networknt.schema.SchemaValidatorsConfig;
+import com.networknt.schema.SpecVersion;
+import com.networknt.schema.ValidationResult;
+import com.networknt.schema.ValidatorTypeCode;
+import com.networknt.schema.uri.URIFactory;
+import com.networknt.schema.uri.URLFactory;
+import com.redhat.cloud.event.core.v1.RHELSystem;
+import com.redhat.cloud.notifications.models.ConsoleCloudEvent;
+import com.redhat.cloud.notifications.validator.LocalDateTimeValidator;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+
+@ApplicationScoped
+public class CloudEventParser {
+
+    private static final String schemaPath = "/schemas/events/v1/events.json";
+
+    static class CloudEventURIFactory implements URIFactory {
+
+        private static final String baseUrl = "https://console.redhat.com/api";
+        private final URLFactory urlFactory = new URLFactory();
+        private final String base;
+
+        CloudEventURIFactory() {
+            String fullPath = RHELSystem.class.getResource(schemaPath).toString();
+            base = fullPath.substring(0, fullPath.length() - schemaPath.length());
+        }
+
+        @Override
+        public URI create(String uri) {
+            return urlFactory.create(replaceBase(uri));
+        }
+
+        @Override
+        public URI create(URI baseURI, String segment) {
+            return urlFactory.create(URI.create(replaceBase(baseURI.toString())), segment);
+        }
+
+        private String replaceBase(String uri) {
+            if (uri.startsWith(baseUrl)) {
+                uri = base + uri.substring(baseUrl.length(), uri.length() - 1);
+            }
+
+            return uri;
+        }
+    }
+
+    @Inject
+    ObjectMapper objectMapper;
+
+    JsonSchema jsonSchema;
+
+    @PostConstruct
+    void init() {
+        jsonSchema = getJsonSchema();
+    }
```

Comment:
Can't this be initialized inline?
```suggestion
    JsonSchema jsonSchema = getJsonSchema();
    }
```
`@PostConstruct` should only be needed when there CDI beans initialization involved.